agent/rpc/peering/service_test.go

```diff
@@ -912,7 +912,7 @@ func newTestServer(t *testing.T, cb func(conf *consul.Config)) testingServer {
 	testrpc.WaitForLeader(t, server.RPC, conf.Datacenter)
 
 	backend := consul.NewPeeringBackend(server, deps.GRPCConnPool)
-	handler := &peering.Service{Backend: backend}
```

Comment:
I think we also need a test in this file to ensure the active peering state shows up. I know you implemented one in the api module but I would also love to see one here in the same package where we actually compute the state.